dv/verilator/pcount/cpp/ibex_pcounts.h

```diff
@@ -0,0 +1,45 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#ifndef IBEX_PCOUNTS_H_
+#define IBEX_PCOUNTS_H_
+
+#include <cstdint>
+#include <string>
+#include <vector>
+
+extern const std::vector<std::string> ibex_count_names;
+
+/**
+ * Given an array of performance counter values returns a formatted string for
+ * printing/logging
+ *
+ * In Ibex the performance counters are held in the mhpmcounter flops in
+ * ibex_cs_registers. In a verilator simulation the signal type of the
+ * mhpmcounter array should be compatible with the type of pcounts here and so
+ * can be passed in directly to this function.
+ *
+ * There are two options for string formatting, csv or pretty-print. Both
+ * produce one counter name and value per line. csv just seperates them with a
+ * comma and no further formatting. pretty-print uses a colon and aligns the
+ * values.
+ *
+ * csv (csv == true):
+ * countername1, 1234
+ * longercountername1, 43980
+ * ...
+ *
+ * pretty-print (csv == false):
+ * countername1:       1234
+ * longercountername1: 43980
+ * ...
+ *
+ * @param pcounts Array of performance counter values, must be of length
+ * ibex_pcount_num
+ * @param csv Choose csv or pretty-print formatting
+ * @return String of formatted performance counter values, newline at end
+ */
+std::string ibex_pcount_string(uint64_t pcounts[], bool csv);
```

Comment:
You could split this into two functions, one for CSV and one for pretty printing. That would make the doxygen string easier to understand, and the implementation more focused. (You can share the lower half of the implementation between the two functions by putting it into a private function.)